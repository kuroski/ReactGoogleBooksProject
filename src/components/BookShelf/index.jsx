import React, {Component} from 'react'
import Book from '../Book'

const propTypes = {
  books: React.PropTypes.array.isRequired,
  onFavorite: React.PropTypes.func.isRequired,
  isOnFavorite: React.PropTypes.func.isRequired
}

class BookShelf extends Component {
  render() {
    return this.props.books.length ? (
      <ul className="BookShelf">
        {this.props.books.map((item, index) => {
          return (
            <Book
              key={index}
              bookId={item.id}
              index={index}
              title={item.volumeInfo.title}
              onFavorite={this.props.onFavorite}
              isOnFavorite={this.props.isOnFavorite}
            />
          )
        })}
      </ul>
    ) : null
  }
}

BookShelf.propTypes = propTypes

export default BookShelf
