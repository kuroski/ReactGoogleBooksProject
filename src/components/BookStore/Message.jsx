import styled from 'styled-components'

const Message = styled.div`
    font-family: Roboto;
    text-align: center;
    background-color: #212121;
    padding: .5rem;
    color: white;

    &:empty {
        display: none;
    }
`

export default Message