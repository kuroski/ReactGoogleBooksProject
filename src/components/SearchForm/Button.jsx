import styled from 'styled-components'

const Button = styled.button`
    position: absolute;
    top: calc(50% - 12px);
    right: .5rem;
    background-color: transparent;
    border: none;
    outline: none;
    color: white;
    cursor: pointer;
`

export default Button