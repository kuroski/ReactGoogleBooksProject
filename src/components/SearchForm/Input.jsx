import styled from 'styled-components'

const Input = styled.input`
    background-color: #6D7993;
    padding: .5rem 1rem;
    flex: 1;
    border: none;
    outline: none;
    color: white;
    font-family: Roboto;
    font-size: 1.3rem;
    box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
`

export default Input