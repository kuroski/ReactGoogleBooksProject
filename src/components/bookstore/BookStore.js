import React, { Component } from 'react';

class BookStore extends Component {
  render() {
    return (
      <div className="BookStore">
        Hello Bookstore
      </div>
    );
  }
}

export default BookStore;
