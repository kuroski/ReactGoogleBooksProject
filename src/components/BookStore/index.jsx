import React, {Component} from 'react'
import SearchForm from '../SearchForm'
import BookShelf from '../BookShelf'
import Pagination from '../Pagination'
import * as booksApi from '../../api'
import {FormattedMessage} from 'react-intl'

class BookStore extends Component {
  constructor(props) {
    super(props)
    this.state = {
      books: [],
      message: '',
      currentSearchTerm: '',
      currentPage: 0,
      pageCount: 0
    }

    this.executeBookSearch = this.executeBookSearch.bind(this)
    this.executePageChange = this.executePageChange.bind(this)
    this.searchAllBooks = this.searchAllBooks.bind(this)
    this.executeFavoriteBook = this.executeFavoriteBook.bind(this)
  }

  executeBookSearch(term) {
    if(!term) {
      this.setState({
        message: <FormattedMessage id="app.errors.nosearchtermprovided" defaultMessage="Nenhum termo de pesquisa foi digitado"></FormattedMessage>
      })
      return
    }

    return this.searchAllBooks(term)
  }

  executePageChange(toPage) {
    return this.searchAllBooks(this.state.currentSearchTerm, toPage)
  }

  searchAllBooks(term, page = 0) {
    return booksApi.all(term, page)
      .then(books => {
        this.setState({
          message: '',
          books: books.items,
          currentSearchTerm: term,
          currentPage: page,
          pageCount: Math.ceil(books.totalItems / 10)
        })
      })
  }

  executeFavoriteBook(bookId) {}

  render() {
    return (
      <div className="BookStore">
        <div>{this.state.message}</div>
        <SearchForm onSubmit={this.executeBookSearch} />
        <BookShelf onFavorite={this.executeFavoriteBook} books={this.state.books} />
        <Pagination currentPage={this.state.currentPage} pageCount={this.state.pageCount} onPageChange={this.executePageChange} />
      </div>
    )
  }
}

export default BookStore
