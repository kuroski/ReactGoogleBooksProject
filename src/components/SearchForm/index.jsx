import React, {Component, PropTypes} from 'react'
import styled from 'styled-components'
import Input from './Input'
import Button from './Button'

const Container = styled.form`
  display: flex;
  text-align: center;
  position: relative;
`;

const propTypes = {
  onSubmit: PropTypes.func.isRequired
}

class SearchForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      term: ''
    }
    this.setTerm = this.setTerm.bind(this)
    this.executeSearch = this.executeSearch.bind(this)
  }

  setTerm(event) {
    this.setState({term: event.target.value})
  }

  executeSearch(event) {
    event.preventDefault()
    this.props.onSubmit(this.state.term)
  }

  render() {
    return (
      <Container>
        <Input autoFocus value={this.state.term} onChange={this.setTerm} />
        <Button onClick={this.executeSearch}>
          <i className="material-icons">search</i>
        </Button>
      </Container>
    )
  }
}

SearchForm.propTypes = propTypes

export default SearchForm