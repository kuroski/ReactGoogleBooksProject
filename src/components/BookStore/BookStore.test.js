import React from 'react'
import BookStore from './'
import SearchForm from '../SearchForm'
import BookShelf from '../BookShelf'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('BookStore', () => {
  it('renders correctly', () => {
    const BookStoreTree = renderer.create(
      <BookStore />
    ).toJSON()
    expect(BookStoreTree).toMatchSnapshot()
  })

  it('should render a SearchForm and a BookShelf', () => {
    const wrapper = shallow(<BookStore />)
    expect(wrapper.containsAllMatchingElements([
      <SearchForm />,
      <BookShelf />
    ])).toEqual(true)
  })

  it('should start with an empty book list', () => {
    const wrapper = shallow(<BookStore />)
    expect(wrapper.state('books')).toEqual([])
  })
})
