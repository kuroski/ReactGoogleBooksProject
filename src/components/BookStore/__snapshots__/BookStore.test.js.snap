exports[`BookStore renders correctly 1`] = `
<div
  className="BookStore">
  <div
    className="SearchForm">
    <input
      onChange={[Function]}
      value="" />
    <button
      onClick={[Function]}>
      Search
    </button>
  </div>
  <div
    className="BookShelf">
    Hello Book Shelf
  </div>
</div>
`;
